src/postgres_client/postgres_client_account_index.rs

```diff
@@ -0,0 +1,419 @@
+        );
+        for j in 0..batch_size {
+            let row = j * TOKEN_INDEX_COLUMN_COUNT;
+            let val_str = format!("(${}, ${}, ${})", row + 1, row + 2, row + 3);
+
+            if j == 0 {
+                stmt = format!("{} {}", &stmt, val_str);
+            } else {
+                stmt = format!("{}, {}", &stmt, val_str);
+            }
+        }
+
+        let handle_conflict =
+            "ON CONFLICT (owner_key, inner_key) DO UPDATE SET slot=excluded.slot where index.slot < excluded.slot";
+
+        stmt = format!("{} {}", stmt, handle_conflict);
+
+        info!("{}", stmt);
+        let bulk_stmt = client.prepare(&stmt);
+
+        match bulk_stmt {
+            Err(err) => {
+                return Err(AccountsDbPluginError::Custom(Box::new(AccountsDbPluginPostgresError::DataSchemaError {
+                    msg: format!(
+                        "Error in preparing for the token owner index update PostgreSQL database: {} host: {:?} user: {:?} config: {:?}",
+                        err, config.host, config.user, config
+                    ),
+                })));
+            }
+            Ok(update_account_stmt) => Ok(update_account_stmt),
+        }
+    }
+
+    /// Build the token mint index bulk insert statement.
+    pub fn build_bulk_token_mint_index_insert_statement(
+        client: &mut Client,
+        config: &AccountsDbPluginPostgresConfig,
+    ) -> Result<Statement, AccountsDbPluginError> {
+        let batch_size = config
+            .batch_size
+            .unwrap_or(DEFAULT_ACCOUNTS_INSERT_BATCH_SIZE);
+        let mut stmt = String::from(
+            "INSERT INTO spl_token_mint_index AS index (mint_key, inner_key, slot) VALUES",
+        );
+        for j in 0..batch_size {
+            let row = j * TOKEN_INDEX_COLUMN_COUNT;
+            let val_str = format!("(${}, ${}, ${})", row + 1, row + 2, row + 3);
+
+            if j == 0 {
+                stmt = format!("{} {}", &stmt, val_str);
+            } else {
+                stmt = format!("{}, {}", &stmt, val_str);
+            }
+        }
+
+        let handle_conflict =
+            "ON CONFLICT (mint_key, inner_key) DO UPDATE SET slot=excluded.slot where index.slot < excluded.slot";
+
+        stmt = format!("{} {}", stmt, handle_conflict);
+
+        info!("{}", stmt);
+        let bulk_stmt = client.prepare(&stmt);
+
+        match bulk_stmt {
+            Err(err) => {
+                return Err(AccountsDbPluginError::Custom(Box::new(AccountsDbPluginPostgresError::DataSchemaError {
+                    msg: format!(
+                        "Error in preparing for the token mint index update PostgreSQL database: {} host: {:?} user: {:?} config: {:?}",
+                        err, config.host, config.user, config
+                    ),
+                })));
+            }
+            Ok(update_account_stmt) => Ok(update_account_stmt),
+        }
+    }
```

Comment:
Same as https://github.com/solana-labs/solana-accountsdb-plugin-postgres/pull/9/files#r815277859, I believe these can be condensed together

Reply:
done